std/array.d

```diff
@@ -3339,6 +3340,69 @@ if (isDynamicArray!A)
     app.put(r[]);
 }
 
+unittest // Issue 13300
+{ 
+    // Issue 13300: appender must work with non pure postblit
+    import std.array : array;
+
+    struct Simple13300
+    {
+        @disable this(); 
+        this(this) { import std.stdio : stdout; stdout; }  // Not pure!
+
+        private:
+        this(int tmp) { }
+    }
+
+    struct Range13300
+    {
+        @property front() { return Simple13300(0); }
+        void popFront() { count++; }
+        @property empty() { return count > 3; }
+        size_t count;
+    }
```

Comment:
I'd suggest that you just use `std.range.range.only` instead and not declare a range type just for this. Something like `assert(only(Simple1330.init, Simple1330.init).array.length == 2);` should work just as well with less code.

Reply:
See comment below